app/models/newsletter_finder.rb

```diff
@@ -0,0 +1,31 @@
+class NewsletterFinder
+  include Enumerable
+
+  delegate :each, to: :newsletters_list
+
+  def initialize(type)
+    @type = type
+  end
+
+  private
+
+  def newsletters_list
+    @newsletters_list ||= newsletters_with_file_extension.map do |newsletter|
+      newsletter.gsub(/\.pdf/, "")
+    end
+  end
+
+  def newsletters_with_file_extension
+    unsorted_newsletters.sort do |date_1, date_2|
+      Date.parse("1-#{date_2}") <=> Date.parse("1-#{date_1}")
```

Comment:
@JessSumner won't this be a problem when they have more than 1 years worth of content?